packages/next/client/components/navigation.ts

```diff
@@ -105,16 +106,44 @@ export function usePathname(): string {
 //   return useContext(LayoutSegmentsContext)
 // }
 
+// TODO-APP: handle parallel routes
+function getSelectedLayoutSegmentPath(
+  tree: FlightRouterState,
+  parallelRouteKey: string,
+  first = true,
+  segmentPath: string[] = []
+): string[] {
+  let node: FlightRouterState
+  if (first) {
+    // Use the provdided parallel route key on the first parallel route
+    node = tree[1][parallelRouteKey]
+  } else {
+    // After first parallel route prefer children, if there's no children pick the first parallel route.
+    const parallelRoutes = tree[1]
+    node = parallelRoutes.children ?? Object.values(parallelRoutes)[0]
+  }
+
+  const segment = node[0]
+  const segmentString = Array.isArray(segment) ? segment[1] : segment
```

Comment:
Can you rename this to `segmentValue`
```suggestion
  const segmentValue = Array.isArray(segment) ? segment[1] : segment
```